src/components/molecules/DayIndicator.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { useContext } from 'react';
+import styled from 'styled-components';
+import ProgressBar from '@/components/atoms/ProgressBar';
+import { MainContext } from '@/contexts/MainContext';
+
+const StyledIndicator = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+
+  .total,
+  .day {
+    margin: 8px 0;
+  }
+`;
+
+interface Props {
+  readonly day: {
+    name: string;
+    index: number;
+  };
+}
+
+const DayIndicator: React.FC<Props> = ({ day }: Props) => {
+  const { transactions } = useContext(MainContext);
+
+  const todayAmount: number = transactions
+    .filter((tx) => new Date(tx.date).getDay() === day.index)
+    .reduce((acc, tx) => acc + tx.price, 0);
+
+  const total: number = +transactions.reduce((acc, tx) => acc + tx.price, 0);
+
+  return (
+    <StyledIndicator>
+      <div className='total'>{`$${todayAmount}`}</div>
+      <ProgressBar percentage={(todayAmount / total) * 100} />
```

Comment:
nie chcesz tego też wyciągnąć do osobnej funkcji? `const currentPercentage = (todayAmount / total) * 100`
ale to raczej pierdoła i może nie warto... a zostawię Ci ten komentarz :D

Reply:
można :D w Reakcie jestem bardziej skłonny do robienia rzeczy inline'owo niż w Vue, chociaż w Vue ostatnio też zacząłem wiecej inline pisać